### ISSUE TITLE
Series FutureWarning in sample.py:152

### ISSUE THREAD

**Issue by sammlapp** [kitzeslab/opensoundscape#884]
FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`

**Comment by LeonardoViotti** [comment]
Managed to reproduce the warning, but not solved it yet. It seems to be a design issue with `AudioSample` or `preprocessor.forward()`.  Here's the repex assuming Powdermill dataset is in a relative path `data/pnre`.

```
from opensoundscape.audio import Audio
from opensoundscape.spectrogram import Spectrogram
from opensoundscape.sample import AudioSample
import matplotlib.pyplot as plt

import torch
from opensoundscape.ml.datasets import AudioFileDataset
from opensoundscape.ml.safe_dataset import SafeDataset
from opensoundscape.preprocess.preprocessors import SpectrogramPreprocessor

import os
import pandas as pd


train_df = pd.DataFrame.from_dict({
    'file' : ['data/pnre/wav_Files/Recording_1/Recording_1_Segment_30.wav', 'data/pnre/wav_Files/Recording_1/Recording_1_Segment_30.wav'],
    'start_time' : [0 , 4],
    'end_time' : [ 4 , 8 ],
    'amro' : [0, 0 ],
    'btnw' : [1, 0]
 }).set_index(['file', 'start_time', 'end_time'])


def collate_audio_samples_to_dict(samples):
    """generate batched tensors of data and labels (in a dictionary)
    
    returns collated samples: a dictionary with keys "samples" and "labels"
    
    assumes that s.data is a Tensor and s.labels is a list/array
    for each sample S
    
    Args:
    
        samples: iterable of AudioSample objects (or other objects
        with attributes .data as Tensor and .labels as list/array)
    
    Returns:
        dictionary of {
            "samples":batched tensor of samples,
            "labels": batched tensor of labels,
        }
    """
    return {
        "samples": torch.stack([s.data for s in samples]),
        "labels": torch.Tensor([s.labels for s in samples]),
    }

def identity(x):
    """return the input unchanged"""
    return x

preprocessor = SpectrogramPreprocessor(sample_duration=4)

train_dataset = AudioFileDataset(train_df, preprocessor)
train_dataset.bypass_augmentations = False
train_safe_dataset = SafeDataset(train_dataset, invalid_sample_behavior="substitute")

train_dataloader_cls = torch.utils.data.DataLoader
train_loader = train_dataloader_cls(
    train_safe_dataset,
    batch_size=2,
    num_workers=0,
    shuffle=True,  # SHUFFLE SAMPLES because we are training
    # use pin_memory=True when loading files on CPU and training on GPU
    pin_memory=False,
    collate_fn=identity,
)


for batch_idx, samples in enumerate(train_loader):
    collate_audio_samples_to_dict(samples)
    break
```

**Comment by LeonardoViotti** [comment]
This issue seems to have been resolved by @louisfh [here](https://github.com/kitzeslab/opensoundscape/blame/883b19988aa7900e2e9fb868d1cc31d7df163277/opensoundscape/sample.py#L160). Should we close this?
